tests/test_registration_retractions.py

```diff
@@ -0,0 +1,217 @@
+             auth=self.auth,
+             expect_errors=True,
+        )
+
+        assert_equal(res.status_code, 400)
+        self.registration.reload()
+        assert_false(self.registration.is_retracted)
+        assert_is_none(self.registration.retracted_justification)
+
+    def test_non_admin_retract_raises_401(self):
+        res = self.app.post_json(self.retraction_url, expect_errors=True)
+
+        assert_equals(res.status_code, 401)
+        self.registration.reload()
+        assert_false(self.registration.is_retracted)
+        assert_is_none(self.registration.retracted_justification)
+
+    def test_retract_without_justification_raises_200(self):
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': ''},
+             auth=self.auth,
+        )
+
+        assert_equal(res.status_code, 200)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        assert_equal(self.registration.retracted_justification, u'')
+
+    def test_redirect_if_retracted(self):
+        expected_redirect_url = self.registration.web_url_for('view_project')
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': self.justification},
+             auth=self.auth,
+             expect_errors=True,
+        )
+
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['redirectUrl'], expected_redirect_url)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        assert_equal(self.registration.retracted_justification, self.justification)
+
+    def test_cant_access_non_approved_resource(self):
+        # Retract public registration
+        expected_redirect_url = self.registration.web_url_for('view_project')
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': self.justification},
+             auth=self.auth,
+             expect_errors=True,
+        )
+        # Verify it's accessible
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['redirectUrl'], expected_redirect_url)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        # Ensure access to resources not explicitly granting permission to retractions returns 400
+        res = self.app.get(
+            self.registration.web_url_for('project_wiki_home'),
+            expect_errors=True
+        )
+        assert_equal(res.status_code, 400)
+
+
+class RegistrationRetractionSearchTestCase(OsfTestCase):
+
+    def test_retract_wiki_not_in_search(self):
+        pass
+
+    def test_retract_search_indicator_in_results(self):
+        pass
+
+
+@must_be_valid_project
+def valid_project_helper(**kwargs):
+    return kwargs
+
+@must_be_valid_project(are_retractions_valid=True)
+def as_factory_allow_retractions(**kwargs):
+    return kwargs
+
+
+class TestValidProject(OsfTestCase):
```

Comment:
Good catch, they were meant to be ported to a more aptly named test file. I was surprised these decorators weren't tested before; they touch **so** many views.